### ISSUE TITLE
`torch.pow` gives incorrect results compared to pure python and `torch.float_power`

### ISSUE THREAD

**Issue by aws-murandoo** [pytorch/xla#5118]
## 🐛 Bug

When doing the following computation: `0.995**100` we see that the following operators produce these results:
```
torch.pow - tensor(0, device='xla:1')
toch.float_power - tensor(0.6058, device='xla:1')
** - 0.6057704364907279
```

The following HLOs are produced for each the first two operators
```
HloModule SyncTensorsGraph.5, entry_computation_layout={(s64[],s64[])->(s64[])}

ENTRY %SyncTensorsGraph.5 (p0: s64[], p1: s64[]) -> (s64[]) {
  %p1 = s64[] parameter(1), frontend_attributes={neff_input_name="input1"}
  %p0 = s64[] parameter(0), frontend_attributes={neff_input_name="input0"}
  %power = s64[] power(s64[] %p1, s64[] %p0)
  ROOT %tuple = (s64[]) tuple(s64[] %power), frontend_attributes={neff_output_names="output0"}
}
```

```
HloModule SyncTensorsGraph.6, entry_computation_layout={(s64[],f32[])->(f32[])}

ENTRY %SyncTensorsGraph.6 (p0: s64[], p1: f32[]) -> (f32[]) {
  %p1 = f32[] parameter(1), frontend_attributes={neff_input_name="input1"}
  %p0 = s64[] parameter(0), frontend_attributes={neff_input_name="input0"}
  %convert = f32[] convert(s64[] %p0)
  %power = f32[] power(f32[] %p1, f32[] %convert)
  ROOT %tuple = (f32[]) tuple(f32[] %power), frontend_attributes={neff_output_names="output0"}
}
```

## To Reproduce

Execute the following script:
```
import torch
import torch_xla.core.xla_model as xm

beta = 0.995
def func1(step):
    step = torch.tensor(step, device=xm.xla_device())
    factor = torch.pow(beta, step)
    print(factor)

def func2(step):
    step = torch.tensor(step, device=xm.xla_device())
    factor = torch.float_power(beta, step)
    factor = factor.float()
    print(factor)

def func3(step):
    factor = (beta ** step)
    print(factor)

func1(100)
func2(100)
func3(100)
```

## Expected behavior

`torch.pow` should produce the same results as CPU python and `torch.float_power`

## Environment

 - Reproducible on XLA backend [CPU/TPU]: Neuron
 - torch_xla version: `1.13.1+torchneuron6`

**Comment by aws-murandoo** [comment]
Same issue occurs on CPU, torch-xla version: `torch-xla                1.13`

**Comment by JackCaoG** [comment]
is it a accuracy problem or number is off too much? What's the cpu result and what if you do `0.995**3` instead of `**100`

**Comment by aws-murandoo** [comment]
It's off too much, `func1(3)` returns 0.

**Comment by aws-murandoo** [comment]
Further context:
Seems to occur only when:

1. The first operand is a Number and the exponent is a Tensor
2. When the first operand is less than 1

See:
```
>>> torch.pow(torch.tensor(0.995, device=xm.xla_device()), 3)
tensor(0.9851, device='xla:0')
>>> torch.pow(0.995, torch.tensor(3))
tensor(0.9851)
>>> torch.pow(0.995, torch.tensor(3, device=xm.xla_device()))
tensor(0, device='xla:0')
>>> torch.pow(torch.tensor(0.995, device=xm.xla_device()), torch.tensor(3, device=xm.xla_device()))
tensor(0.9851, device='xla:0')
>>> torch.pow(2, torch.tensor(3, device=xm.xla_device()))
tensor(8, device='xla:0')
>>> torch.pow(1, torch.tensor(3, device=xm.xla_device()))
tensor(1, device='xla:0')
>>> torch.pow(0.5, torch.tensor(3, device=xm.xla_device()))
tensor(0, device='xla:0')
```

**Comment by JackCaoG** [comment]
@lsy323 will take a look this week.

**Comment by aws-murandoo** [comment]
@lsy323 Any update on this one?

**Comment by JackCaoG** [comment]
@lsy323 Do you have cycle to look at this one? This is blocking a couple other pytorch pr? Let me know if you need some help.

**Comment by lsy323** [comment]
Sorry, I was pretty loaded recently, I will take a look tomorrow morning.
